src/bcf/report/table_report/alignment_reader.rs

```diff
@@ -0,0 +1,501 @@
+}
+
+pub fn read_indexed_bam(path: &Path, chrom: String, from: u64, to: u64) -> Vec<Alignment> {
+    let mut bam = bam::IndexedReader::from_path(&path).unwrap();
+    let tid = bam.header().tid(chrom.as_bytes()).unwrap();
+
+    let mut alignments: Vec<Alignment> = Vec::new();
+
+    bam.fetch(tid, from, to).unwrap();
+
+    for r in bam.records() {
+        let rec = r.unwrap();
+
+        let a = make_alignment(&rec);
+
+        alignments.push(a);
+    }
+
+    alignments
+}
+
+fn make_alignment(record: &bam::Record) -> Alignment {
+    let has_pair = record.is_paired();
+
+    let mate_pos = record.mpos();
+
+    let tid = record.tid();
+    let mtid = record.mtid();
+
+    //Cigar String
+    let cigstring = record.cigar();
+
+    //Position
+    let pos = record.pos();
+
+    //Länge
+    let le = record.seq().len() as u16;
+
+    //Sequenz
+    let seq = record.seq().as_bytes();
+    let sequenz = String::from_utf8(seq).unwrap();
+
+    //Flags
+    let flgs = record.flags();
+    let flag_string = decode_flags(flgs);
+
+    //Name
+    let n = record.qname().to_owned();
+    let name = String::from_utf8(n).unwrap();
+
+    let read = Alignment {
+        sequence: sequenz,
+        pos: pos,
+        length: le,
+        cigar: cigstring,
+        flags: flag_string,
+        name: name,
+        paired: has_pair,
+        mate_pos: mate_pos,
+        tid: tid,
+        mate_tid: mtid,
+    };
+
+    read
+}
+
+pub fn make_nucleobases(
+    fasta_path: &Path,
+    chrom: String,
+    snippets: Vec<Alignment>,
+    from: u64,
+    to: u64,
+) -> (Vec<AlignmentNucleobase>, Vec<AlignmentMatch>) {
+    let mut bases: Vec<AlignmentNucleobase> = Vec::new();
+    let mut matches: Vec<AlignmentMatch> = Vec::new();
+
+    let ref_bases = read_fasta(fasta_path, chrom, from, to);
+
+    for s in snippets {
+        let mut cigar_offset: i64 = 0;
+        let mut read_offset: i64 = 0;
+        let base_string = s.sequence.clone();
+        let char_vec: Vec<char> = base_string.chars().collect();
+
+        let mut soft_clip_begin = true;
+
+        let p = s.clone();
+
+        if p.paired && (p.pos + p.length as i64) < p.mate_pos && p.tid == p.mate_tid {
+            let pairing = AlignmentMatch {
+                marker_type: Marker::Pairing,
+                start_position: (p.pos + p.length as i64) as f64 - 0.5,
+                end_position: p.mate_pos as f64 - 0.5,
+                flags: p.flags.clone(),
+                name: p.name.clone(),
+                read_start: p.pos.clone() as u32,
+                read_end: (p.mate_pos.clone() + 100) as u32,
+            };
+
+            matches.push(pairing);
+        }
+
+        for c in s.cigar.iter() {
+            let mut match_count = 0;
+            let mut match_start = 0;
+            let mut match_ending = false;
+
+            match c {
+                rust_htslib::bam::record::Cigar::Match(c) => {
+                    for _i in 0..rust_htslib::bam::record::Cigar::Match(*c).len() {
```

Comment:
Please move the bodies of this match into separate functions. That would make this function easier to read.